controllers/users.js

```diff
@@ -462,6 +464,46 @@ const filterUsers = async (req, res) => {
   }
 };
 
+// one time script function to perform the migration - adding github_user_id field to the document
+const migrate = async (req, res) => {
+  try {
+    // Fetch user data from GitHub API for each document in the users collection
+    // divided by 500 because firestore api guarantee that we can process in batch of 500.
+    const usersSnapshot = await firestore.collection("users").get();
+    const batchCount = Math.ceil(usersSnapshot.docs.length / 500);
+    // Create batch write operations for each batch of documents
+    for (let i = 0; i < batchCount; i++) {
+      const batchDocs = usersSnapshot.docs.slice(i * 500, (i + 1) * 500);
+      const batchWrite = firestore.batch();
+      const batchWrites = [];
+      for (const userDoc of batchDocs) {
+        const githubUsername = userDoc.data().github_id;
+        batchWrite.update(userDoc.ref, { github_user_id: null });
+        batchWrites.push(
+          axios
+            .get(`https://api.github.com/users/${githubUsername}`)
```

Comment:
We are using the unauthenticated way to call the GitHub API and it is rate limited to 60 requests per hour.
Then, won't this call start failing once we reach that limit? 
Maybe I am missing something

Reply:
https://island-macrame-1be.notion.site/Architecture-Changes-062bfeb3d2114bacaddc1d9cdafe166e 
please refer this doc.

Reply:
@iamYashSinha your code is right here or @shubham-y is right here?? Can you share the conclusion??